packages/react-native/native-js-interface.md

```diff
@@ -4,27 +4,143 @@ This is a working spec of this the interface that the native layer (`BugsnagReac
+
+An "empty" map will indicate to the JS layer that the native client is not configured, which is a problem.
+
+This method is required to be synchronous so it should be annotated as such. This is because during its synchronous configuration, the JS layer needs to obtain config from the native layer.
+
+###### Android
+
+```java
+@ReactMethod(isBlockingSynchronousMethod = true)
+public ReadableMap getConfig()
+```
+
+###### iOS
+```objc
+(NSDictionary *)getConfig;
+
+// annotation goes in BugsnagReactNative.m
+RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD(getConfig);
+```
+
+#### `BugsnagReactNative.updateClientState(key, updates): void`
+
+Updates to mutable client state in JS (e.g. `metaData`, `user`) should be replicated and stored in the native layer using this method.
+
+###### Android
+```java
+void updateClientState(String key, ReadableMap update)
+```
+
+###### iOS
+```objc
+- (void)updateClientState:(NSString * key)
+                         :newValue (NSDictionary *)update;
+```
+
+The keys in the updates map are the names of the properties to be updated and the values are encoded objects containing the new value for that property. The list of allowed keys is:
+
+- `user`: a Map<String, String> with `id`, `name` and `email`
+- `metaData`: a Map<String, Map> of metadata tabs
+
+#### `BugsnagReactNative.deliver(payload): Promise<boolean>`
+
+Send a report that was created in JS-land. The boolean return value denotes the following:
+
+- `true`: the report was sent, or was enqueued with the intention of sending later
+- `false`: the report was not sent and will never be sent
+
+The `report` contains this subset of the error reporting payload structure:
+
+| Key                        | Type                  |
+|----------------------------|-----------------------|
+| `exceptions`               | `Array[Exception]`    |
+| `severity`                 | `String`              |
+| `unhandled`                | `boolean`             |
+| `app`                      | `Map<String, String>` |
+| `device`                   | `Map<String, String>` |
+| `breadcrumbs`              | `Array<Breadcrumb>`   |
+| `context`                  | `String`              |
+| `user`                     | `Map`                 |
+| `metaData`                 | `Map<String, Map>`    |
+| `groupingHash`             | `String`              |
+
+The native layer will be responsible for tracking the handled/unhandled count based on the handled state in the provided payload, and for using this payload as a seed for constructing the actual payload it will send.
+
+###### Android
+```java
+public void deliver(ReadableMap payload, Promise promise)
+```
+
+###### iOS
+```objc
+- (void)deliver:(NSDictionary *)payload
+        resolve:(RCTPromiseResolveBlock)resolve
+         reject:(RCTPromiseRejectBlock)reject;
+```
+
+#### `BugsnagReactNative.getPayloadInfo(): Promise<info>`
+
+Retrieves info from the native layer so that JS `beforeSend` callbacks can access it.
```

Comment:
Will this native info be combined with the JS info before being exposed in a callback?